packages/js/src/plugins/auctionHouseModule/operations/withdrawFromBuyerAccount.ts

```diff
@@ -0,0 +1,222 @@
+import { ConfirmOptions, PublicKey } from '@solana/web3.js';
+import type { Metaplex } from '@/Metaplex';
+import { TransactionBuilder } from '@/utils';
+import {
+  WithdrawInstructionAccounts,
+  createWithdrawInstruction,
+  createAuctioneerWithdrawInstruction,
+} from '@metaplex-foundation/mpl-auction-house';
+import {
+  useOperation,
+  Operation,
+  OperationHandler,
+  Signer,
+  isSigner,
+  toPublicKey,
+  SplTokenAmount,
+  SolAmount,
+} from '@/types';
+import { SendAndConfirmTransactionResponse } from '../../rpcModule';
+import { AuctionHouse } from '../models';
+import { findAuctioneerPda, findAuctionHouseBuyerEscrowPda } from '../pdas';
+import { AuctioneerAuthorityRequiredError } from '../errors';
+
+// -----------------
+// Operation
+// -----------------
+
+const Key = 'WithdrawFromBuyerAccountOperation' as const;
+
+/**
+ * Withdraws funds from the user's buyer escrow account for the given auction house.
+ *
+ * ```ts
+ * await metaplex
+ *   .auctionHouse()
+ *   .withdraw({ auctionHouse, buyer, amount })
+ *   .run();
+ * ```
+ *
+ * @group Operations
+ * @category Constructors
+ */
+export const withdrawFromBuyerAccountOperation = useOperation<WithdrawOperation>(Key);
+
+/**
+ * @group Operations
+ * @category Types
+ */
+export type WithdrawOperation = Operation<
+  typeof Key,
+  WithdrawInput,
+  WithdrawOutput
+>;
+
+/**
+ * @group Operations
+ * @category Inputs
+ */
+export type WithdrawInput = {
+  /** The Auction House from which escrow buyer withdraws funds. */
+  auctionHouse: Pick<
+    AuctionHouse,
+    | 'address'
+    | 'authorityAddress'
+    | 'hasAuctioneer'
+    | 'treasuryMint'
+    | 'feeAccountAddress'
+    >;
+  /**
+   * The buyer who withdraws funds.
+   * This expects a Signer.
+   *
+   * @defaultValue `metaplex.identity()`
+   */
+  buyer?: PublicKey | Signer;
+  /**
+   * The Authority key.
+   * It is required when the buyer is not a signer.
+   *
+   * @defaultValue Defaults to not being used.
+   */
+  authority?: PublicKey | Signer;
```

Comment:
If it defaults to not being used, then why do we need it as a PublicKey sometimes? If `authority` is a public key, then we already have this information under `auctionHouse.authorityAddress` right?

Reply:
Right, authority can be removed here

Reply:
I think you still need the option to pass it as a `Signer` but not as `PublicKey | Signer` since the `PublicKey` part is redundant.

Reply:
Either authority or buyer must be a signer actually.

Here goes the rust code

```
if !wallet.to_account_info().is_signer && !authority.to_account_info().is_signer {
        return Err(AuctionHouseError::NoValidSignerPresent.into());
    }
```

Reply:
So, you must do it the same way as in createListing.ts.

```
  // Update the account to be a signer since it's not covered properly by MPL due to its dynamic nature.
  const signerKeyIndex = sellInstruction.keys.findIndex((key) =>
    key.pubkey.equals(signer.publicKey)
  );
  sellInstruction.keys[signerKeyIndex].isSigner = true;
```